### PR TITLE
feature: baremetal colloct isolated devices info

### PR DESCRIPTION
**Pull request by wanyaoqi**
**这个 PR 实现什么功能/修复什么问题**:
```
feature: baremetal colloct isolated devices info
```

**是否需要 backport 到之前的 release 分支**:
NONE
/area region host baremetal
/cc @swordqiu @zexi

### CODE REVIEW

File: pkg/compute/tasks/baremetal_delete_task.go [hunk 1]
```diff
@@ -53,11 +53,20 @@ func (self *BaremetalDeleteTask) OnInit(ctx context.Context, obj db.IStandaloneM
...
 func (self *BaremetalDeleteTask) OnDeleteBaremetalComplete(ctx context.Context, baremetal *models.SHost, body jsonutils.JSONObject) {
+	// clean isolated device recoreds
+	devs := models.IsolatedDeviceManager.FindByHost(baremetal.GetId())
+	for i := 0; i < len(devs); i++ {
+		if err := devs[i].ValidateDeleteCondition(ctx); err != nil {
+			self.OnFailure(ctx, baremetal, jsonutils.NewString(err.Error()))
+			return
+		}
+		devs[i].CustomizeDelete(ctx, self.UserCred, nil, nil)
+	}
```
Comment (swordqiu) [inline pkg/compute/tasks/baremetal_delete_task.go:69]: 这部分逻辑重复了

**Comment by swordqiu** [comment]
1. host spec should including GPU-spec
2. scheduler should considering baremetal GPU-constraints
3. the GPU on baremetal should not be accessible to ordinary VM

**Comment by wanyaoqi** [comment]
> 1. host spec should including GPU-spec
> 2. scheduler should considering baremetal GPU-constraints
> 3. the GPU on baremetal should not be accessible to ordinary VM

done

**Pull request merged**